iree/turbine/kernel/wave/templates/speculative_decoding.py

```diff
@@ -124,14 +124,16 @@ def tree_speculative_sampling(
             coin * sum_relu, target_shape=[BATCH_SIZE, NUM_DRAFT_TOKENS, VOCAB_SIZE]
         )
         greater_than_u = cdf > threshold_dist_u
-        pad_token = tkl.Register[BATCH_SIZE, NUM_DRAFT_TOKENS, VOCAB_SIZE, tkl.i32](1e6)
+        pad_token = tkl.Register[BATCH_SIZE, NUM_DRAFT_TOKENS, VOCAB_SIZE, tkl.i32](
```

Comment:
nice find. can we add a comment why it should be `VOCAB_SOZE - 1`?

Reply:
I've added it right below